File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        int result = 0;
        double exponent = Math.Floor(Math.Log10(number) + 1);
        foreach (int digit in DigitsReversed(number))
        {
            result += (int)Math.Pow(digit, exponent);
        }
        return result == number;
    }
    private static IEnumerable<int> DigitsReversed(int number)
    {
        while (number != 0)
        {
            yield return number % 10;
            number /= 10;
        }
    }
}

File: fsharp/armstrong-numbers/ArmstrongNumbers.fs
module ArmstrongNumbers

open System

let digitsReversed (number: int): int seq = 
    let mutable temp = number
    seq {while temp <> 0 do
            yield temp % 10
            temp <- temp / 10 }

let isArmstrongNumber (number: int): bool = 
    let exponent = Math.Floor(Math.Log10(number |> double) + 1.)
    digitsReversed number |> Seq.sumBy (fun elem -> Math.Pow(elem |> double, exponent) |> int) = number

